test/gtest/uct/test_peer_failure.cc

```diff
@@ -112,21 +112,33 @@ ucs_status_t test_uct_peer_failure::err_cb(void *arg, uct_ep_h ep,
     }
 }
 
-void test_uct_peer_failure::kill_receiver(unsigned sender_idx)
+void test_uct_peer_failure::inject_error(unsigned sender_idx)
 {
-    ucs_assert(!m_receivers.empty());
-
     uct_ep_h ep         = m_sender->ep(sender_idx);
-    ucs_status_t status = ep->iface->ops.ep_invalidate(ep, 0);
+    ucs_status_t status = uct_ep_invalidate(ep, 0);
+
     if (status == UCS_ERR_UNSUPPORTED) {
-        m_entities.remove(m_receivers.front());
-        ucs_assert(m_entities.size() == m_receivers.size());
-        m_receivers.erase(m_receivers.begin());
+        kill_receiver(sender_idx);
     } else {
         ASSERT_UCS_OK(status);
     }
 }
 
+void test_uct_peer_failure::kill_receiver(unsigned idx)
+{
+    static unsigned check_idx = 0;
+
+    if (idx == 0) {
+        check_idx = 0;
+    }
```

Comment:
it won't work, in this case index always should be 1, (0 is sender entity). Keeping invalid entitiy is not good idea, IMO

Reply:
can we use m_entities.size() instead of check_idx? seems wrong to track index by a global var

Reply:
it's already below, see L136,138 but it does not rely on exact index
we can add new class member variable to track this but logic won't be simple.
IMO better to refactor entities and setup exact relation sender->seceiver in separate PR to avoid changing lib and test code significantly in the same PR, WDYT?